core/src/impl/Kokkos_Abort.cpp

```diff
@@ -0,0 +1,47 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_IMPL_PUBLIC_INCLUDE
+#define KOKKOS_IMPL_PUBLIC_INCLUDE
+#endif
+
+#include <cstdlib>
+#include <iostream>
+#include <Kokkos_Abort.hpp>
+#include <impl/Kokkos_Error.hpp>
```

Comment:
What do we use from that header?

Reply:
`<iostream>`: `std::cerr`
`<cstdlib>`: `::abort`
I'll remove `<impl/Kokkos_Error.hpp>`.

Reply:
I was specifically asking about `<impl/Kokkos_Error.hpp>`.

No opinion about where to put `Impl::{host,device}_abort`.